src/state/compact-header.mixin.ts

```diff
@@ -0,0 +1,31 @@
+import { HASSDomEvent } from "../common/dom/fire_event";
+import { Constructor, HomeAssistant } from "../types";
+import type { PropertyValues } from "lit-element";
+import { storeState } from "../util/ha-pref-storage";
+import { HassBaseEl } from "./hass-base-mixin";
+
+interface CompactHeaderParams {
+  compact: HomeAssistant["compactHeader"];
+}
+
+declare global {
+  // for fire event
+  interface HASSDomEvents {
+    "hass-compact-header": CompactHeaderParams;
+  }
+  // for add event listener
+  interface HTMLElementEventMap {
+    "hass-compact-header": HASSDomEvent<CompactHeaderParams>;
+  }
+}
+
+export default <T extends Constructor<HassBaseEl>>(superClass: T) =>
```

Comment:
This does not need to be it's own mixin

Reply:
I'm still uncertain on mixins and the best way to add this, will need some guidance/feedback.